### DISCUSSION TITLE
Proxy.py for Android

### DISCUSSION THREAD

**Discussion by koochack** [abhinavsingh/proxy.py#1472] [Q&A] [answered]
Can proxy.py be run on Android? Any advice and general guidelines on how to do this would be appreciated. I assume for Android it must become a VPN service right? Any modifications needed to the code to make it Android compatible? Which tools to use for bringing a Python project into Android? Any other tips or advice? Thanks in advance!

**Comment by abhinavsingh** [comment]
@koochack I asked ChatGPT and it told https://chatgpt.com/share/650ac8bc-d4f8-4713-89d9-6f72a392bae5 .  Give it a try please, I haven't tried running `proxy.py` on Android myself, but I see no reason why we can't 🗡️   Let me know

**Comment by koochack** [comment, marked as answer]
Hey so I tried this and was successfully able to run the proxy server
inside the virtual environment on my Android. But the problem is it only
runs within the termux environment. We need the android to recognize that
server and route all traffic through the proxy. Even if with some complex
trick we manage to do that, I think the added layer of the virtual
environment will tremendously affect the traffic speed and latency.

Edit:
To my surprise, just setting proxy address and port on the Android's wifi connection _does_ route traffic through the server listening inside Termux. I did NOT expect this to work! But hey, can't complain haha
